### ISSUE TITLE
System status

### ISSUE THREAD

**Issue by gclitheroe** [GeoNet/mtr#149]
Is some sort of per metric heartbeat better?  Discuss with @dc735 and @ozym

**Comment by gclitheroe** [comment]
Had some useful discussion with @quiffman and @dc735.

* we agreed that users being able to trust the metrics system is vital
* we should try to minimize system breakages (although they will happen).
* we do want some sort of system insight for the metrics users e.g., "when something looks wrong can we tell if the mtr system is still working but the links are failing etc).
* purple is a pretty confusing way to do this and seems like a bit of a kludge in Xymon / BB etc.

We came up with the idea of adding to metrics a collector agent heart beat and possibly the list of things that each agent is responsible for.  Then we can maybe see if an agent is failing to heartbeat and that may affect all the things it collects for.

@ozym - meant to talk about this in the Tuesday meeting.  Can chat next Tuesday or here if you have ideas or comments.

**Comment by ozym** [comment]
Yes agreed, in some ways the connection test is supposed to be like a heartbeat. It should return a bad result (0 at the moment) if it couldn't connect etc. The problem here might be if there are multiple collection routines doing different parts, at the moment this is only ping vs the rest. I had thought of merging the ping in - but it has requirements that it needs to be run as root (to use raw sockets) so it impacts the other tests (plus I had the idea we run the ping tests more often than the other tests).

In some cases I get an error message back (i.e. "Cannot See", "Cannot See") which I could pass back along with the result (although this is veering into logging).
